apps/protocol-api/src/prisma/resolvers/Contribution.ts

```diff
@@ -535,33 +533,24 @@ export class ContributionCustomResolver {
     // guild_id is only present on the guild contribution table
     // need to run a join between contribution + guild contribution if guild_id
     // is supplied
-    console.log(args.where);
-    const guildIds = args.where?.guildIds;
-
-    if (!guildIds || guildIds.length === 0) {
-      const result = await prisma.$queryRaw<ContributionCountByDate>`
-        SELECT date(date_of_engagement), count(date_of_engagement) as count
-        FROM "Contribution"
-        WHERE ("Contribution"."date_of_engagement" BETWEEN ${start} AND ${end}
-        AND "Contribution"."user_id" = ${user_id})
-        GROUP BY date(date_of_engagement)
-        ORDER BY date(date_of_engagement);`;
-      return result;
-    }
-
-    const result = await prisma.$queryRaw<ContributionCountByDate>`
-      SELECT gc.guild_id, g.name, date(date_of_engagement), count(date_of_engagement) as count
-      FROM "Contribution"
-      INNER JOIN "GuildContribution" as gc
-      ON "Contribution"."id" = gc."contribution_id"
-      INNER JOIN "Guild" as g
-      ON g."id" = gc."guild_id"
-      WHERE ("Contribution"."date_of_engagement" BETWEEN ${start} AND ${end}
-      AND "Contribution"."user_id" = ${user_id}
-      AND gc."guild_id" in (${Prisma.join(guildIds)}))
+    const guildIds = args.where?.guildIds ?? [-1];
+
+    return await prisma.$queryRaw<ContributionCountByDate>`
+      SELECT gc.guild_id,
+             coalesce(g.name, 'Unassigned') as name,
+             date(date_of_engagement),
+             count(date_of_engagement)      as count
+      FROM "Contribution" c
+             LEFT JOIN "GuildContribution" as gc
+                       ON c."id" = gc."contribution_id"
+             LEFT JOIN "Guild" as g
+                       ON g."id" = gc."guild_id"
+      WHERE (c."date_of_engagement" BETWEEN ${start} AND ${end}
+        AND c."user_id" = ${user_id}
+        AND (NOT c.id IN (SELECT contribution_id as id FROM "GuildContribution") OR gc."guild_id" in (${Prisma.join(
```

Comment:
Why do we need the subquery? Also couldn't the or be a conditional string that is added if there are guild ids? @amrro

Reply:
> Why do we need the subquery?

We need to count the contributions that doesn't belong to any guild. So, the subquery loads all `contribution_id`s and counts the ones that does `NOT` belong to it. 

> Also couldn't the or be a conditional string that is added if there are guild ids?

Yes, we can. Is this going to impact performance?

Reply:
True, but wouldn't it be more effcient to do an OR gc."guild_id" is NULL?